src/main/scala/akka/io/AsyncDnsResolver.scala

```diff
@@ -149,9 +182,42 @@ object AsyncDnsResolver {
 
   def parseNameserverAddress(str: String): InetSocketAddress = {
     val inetSocketAddress(host, port) = str
-    new InetSocketAddress(host, Option(port).map(_.toInt).getOrElse(53))
+    new InetSocketAddress(host, Option(port).fold(53)(_.toInt))
   }
 
   private[io] def isSrv(name: String) =
     name.nonEmpty && name(0) == '_'
+
+  private val ipv4Address = """^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$""".r
+
+  private[io] def isInetAddress(name: String): Boolean =
+    (name.contains(":") && !name.contains(".")) || ipv4Address.findAllMatchIn(name).nonEmpty
+
```

Comment:
New code: quick and dirty (but should be reliable) check for whether an address is IPv4 or IPv6.